fvtest/porttest/si.cpp

```diff
@@ -796,6 +796,106 @@ TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_CORE_FILE)
 	reportTestExit(OMRPORTLIB, testName);
 }
 
+	/**
+	 *
+	 * Test omrsysinfo_test_sysinfo_set_limit and omrsysinfo_test_sysinfo_get_limit
+	 * with resourceID OMRPORT_RESOURCE_FILE_DESCRIPTORS
+	 *
+	 */
+	TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_FILE_DESCRIPTORS)
+	{
+		OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+		const char *testName = "omrsysinfo_test_sysinfo_set_limit_FILE_DESCRIPTORS";
+		intptr_t rc = -1;
+		uint64_t originalCurLimit = 0;
+		uint64_t originalMaxLimit = 0;
+		uint64_t currentLimit = 0;
+		const uint64_t descriptorLimit = 256;
+
+		reportTestEntry(OMRPORTLIB, testName);
+
+		/* save original soft limit */
+		omrsysinfo_get_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, &originalCurLimit);
+
+		rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, descriptorLimit);
+		if (0 != rc) {
+			outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit FAILED\n");
+			reportTestExit(OMRPORTLIB, testName);
+			return;
+		}
+
+		omrsysinfo_get_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, &currentLimit);
+		if (descriptorLimit == currentLimit) {
+			portTestEnv->log("omrsysinfo_set_limit set FILE_DESCRIPTORS soft max successful\n");
+		} else {
+			outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit set FILE_DESCRIPTORS soft max FAILED\n");
+			reportTestExit(OMRPORTLIB, testName);
+			return;
+		}
+
+		/* save original hard limit */
+		omrsysinfo_get_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, &originalMaxLimit);
+
+		/* lowering the hard limit is irreversible unless privileged */
+		if (0 != geteuid()) {
+			/* we should be able to set the hard limit to the current value as an unprivileged user
+			   When the hard limit is set to unlimited (-1) a regular user can change it to
+			   any value, but not back to unlimited. */
+			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, originalMaxLimit);
```

Comment:
Assigning the current value back won't tell you if the set was successful or not.  It would have to attempt to set a different value to be an accurate test.